### PR TITLE
Adding debugging and print statements all over

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

**Comment by cwgoes** [comment]
Looks like this doesn't build?

File: src/Juvix/Backends/LLVM/Codegen/Graph.hs [hunk 1]
```diff
@@ -1,723 +1,12 @@
...
-        Types.address' = bothPrimary,
-        Types.indincies' = Block.constant32List [0, 1]
-      }
-
-loadPrimaryNode ∷ RetInstruction m ⇒ Type.Type → Operand.Operand → m Operand.Operand
-loadPrimaryNode nodePtrType e = getPrimaryNode nodePtrType e >>= load nodePtrType
+-- |
+-- Serves as a front end for the Graph folder
+module Juvix.Backends.LLVM.Codegen.Graph
+  ( module Juvix.Backends.LLVM.Codegen.Graph.Definitons,
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Codegen/Graph.hs]:
```suggestion
  ( module Juvix.Backends.LLVM.Codegen.Graph.Definitions,
```

File: src/Juvix/Backends/LLVM/Codegen/Graph.hs [hunk 2]
```diff
@@ -1,723 +1,12 @@
...
+    module Juvix.Backends.LLVM.Codegen.Graph.Operations,
+    module Juvix.Backends.LLVM.Codegen.Graph.Debug,
+  )
+where
+
+import Juvix.Backends.LLVM.Codegen.Graph.Debug
+import Juvix.Backends.LLVM.Codegen.Graph.Definitons
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Codegen/Graph.hs]:
```suggestion
import Juvix.Backends.LLVM.Codegen.Graph.Definitions
```

File: src/Juvix/Backends/LLVM/Codegen/Graph/Definitons.hs [hunk 3]
```diff
@@ -0,0 +1,458 @@
+-- | Operations necessary to update nodes
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Codegen/Graph/Definitons.hs]: I presume this file was moved? Seems Git doesn't pick it up in the diff form.

Comment (cwgoes) [reply src/Juvix/Backends/LLVM/Codegen/Graph/Definitons.hs]: The debugging info looks fine though.

Comment (mariari) [reply src/Juvix/Backends/LLVM/Codegen/Graph/Definitons.hs]: Yeah this file was git mv'd to minimize the git diff

**Comment by cwgoes** [comment]
@mariari Still doesn't build, I think you need additional capability constraints on some of the functions.

**Comment by mariari** [comment]
> @mariari Still doesn't build, I think you need additional capability constraints on some of the functions.

Should build now

**Pull request merged**